library/src/scala/tasty/reflect/Core.scala

```diff
@@ -311,6 +311,9 @@ trait Core {
     /** Type tree representing a type bound written in the source */
     type TypeBoundsTree <: TypeOrBoundsTree
 
+    /** Type tree representing wildcard type bounds written in the source */
+    type WildcardType <: TypeOrBoundsTree
+
```

Comment:
I think we can give the [hypothesis](https://github.com/lampepfl/dotty/pull/5483#issuecomment-440605499) in #5483 a second thought.

Even if we want to report errors for type trees:

- it's completely acceptable for end users to report error for the _whole_ type tree
- they work for type alias and infered types as well
- In practice, I conjecture that macro authors will check types to detect errors, and then just report errors on the _whole_ type tree instead of inspecting the components of type trees & deal with inferred and aliased types.

```Scala
fooMacro[List[Int]]
              ^^^
```